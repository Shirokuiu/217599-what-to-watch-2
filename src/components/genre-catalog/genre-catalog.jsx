import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import GenreCatalogTab from "../genre-catalog-tab/genre-catalog-tab";
import LoadMore from "../load-more/load-more";

import {MovieList} from "../movie-list/movie-list";

export const GenreCatalog = (props) => {
  const {onMovieClick, movies, moviesLoadedCount} = props;

  return <section className="catalog">
    <h2 className="catalog__title visually-hidden">Catalog</h2>

    <GenreCatalogTab />

    <MovieList films={movies} onMovieClick={onMovieClick}/>

    {movies.length >= moviesLoadedCount ? <LoadMore /> : null}
  </section>;
};

GenreCatalog.propTypes = {
  onMovieClick: PropTypes.func.isRequired,
  movies: PropTypes.array.isRequired,
  moviesLoadedCount: PropTypes.number
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  movies: state.movie.genreCatalog.movies,
  moviesLoadedCount: state.movie.genreCatalog.moviesLoadedCount
});

export default connect(mapStateToProps)(GenreCatalog);

