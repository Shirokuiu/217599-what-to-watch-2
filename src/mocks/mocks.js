export const films = [
  {
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Fantastic Beasts`
  },
  {
    img: `img/bohemian-rhapsody.jpg`,
    title: `Bohemian Rhapsody`
  },
  {
    img: `img/macbeth.jpg`,
    title: `Macbeth`
  },
  {
    img: `img/aviator.jpg`,
    title: `Aviator`
  },
  {
    img: `img/we-need-to-talk-about-kevin.jpg`,
    title: `We need to talk about Kevin`
  }
];
