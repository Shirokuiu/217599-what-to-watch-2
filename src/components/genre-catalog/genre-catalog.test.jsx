import React from "react";
import Enzyme, {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import {GenreCatalog} from "../genre-catalog/genre-catalog";

import {films} from "../../mocks/mocks";

Enzyme.configure({adapter: new Adapter()});

it(`Is genre catalog rendered`, () => {
  const tree = shallow(<GenreCatalog
    movies={films}
    activeTab={`All genres`}
    moviesLoadedCount={9}
    onTabClick={jest.fn()}
    onLoadMoreClick={jest.fn()}
    onGetMoviesByGenre={jest.fn()}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});
