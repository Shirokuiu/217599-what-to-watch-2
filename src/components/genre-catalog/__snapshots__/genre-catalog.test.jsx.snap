// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is genre catalog rendered 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <Connect(GenreCatalogTab)
    activeTab="All genres"
    onTabClick={[MockFunction]}
  />
  <MovieList
    movies={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": false,
          "name": "The Grand Budapest Hotel",
          "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
          "previewImage": "img/the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://some-link",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
          "videoLink": "https://some-link",
        },
      ]
    }
  />
</section>
`;
